pkg/operator/ceph/pool/controller.go

```diff
@@ -181,6 +182,11 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 			return reconcile.Result{}, errors.Wrapf(err, "failed to delete pool %q. ", cephBlockPool.Name)
 		}
 
+		// disable RBD stats collection if cephBlockPool was deleted
+		if err := configureRBDStats(r.context, clusterInfo); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to disable stats collection for pool(s)")
```

Comment:
What is the consequence if the pool is not removed from the list for stats? Is there some bad side effect? If not, I would suggest we just log the error and allow the pool finalizer to be removed instead of blocking the removal.

Reply:
I don't think there will be any side effect. And, deleted pool will be removed during next reconcile anyway.
Going ahead with just logging error during Pool deletion.